File: hackerrank/easy/0036-sherlock-and-squares/python/solution.py
## solution.py

import math

def squares(a: int, b: int) -> int:
    start = math.ceil(math.sqrt(a))
    end = math.floor(math.sqrt(b))
    return max(0, end - start + 1)


# --- demo runs ---
if __name__ == "__main__":
    print(squares(3, 9))     # 2
    print(squares(17, 24))   # 0
    print(squares(1, 1))     # 1

File: hackerrank/easy/0036-sherlock-and-squares/js/solution.mjs
// solution.mjs

export function squares(a, b) {
  const start = Math.ceil(Math.sqrt(a));
  const end = Math.floor(Math.sqrt(b));
  return Math.max(0, end - start + 1);
}

// --- demo runs ---
console.log(squares(3, 9));    // 2
console.log(squares(17, 24));  // 0
console.log(squares(1, 1));    // 1